vars/stageStatusFilename.groovy

```diff
@@ -0,0 +1,23 @@
+// vars/stageStatusFilename.groovy
+
+import java.net.URLEncoder
+
+@NonCPS
+String urlEncode() {
+    return URLEncoder.encode(env.STAGE_NAME, 'UTF-8')
+}
+
+  /**
+   * stageStatusFilename step method
+   *
+   * @param config Map of parameters passed, Unused
+   * returns: String with Url Encoded filename for stage name status.
+   */
+
+String call(Map config= [:]) {
+    String directory = 'stage_status'
+    if (!fileExists(directory)) {
+        fileOperations([folderCreateOperation(directory)])
+    }
+    return directory + '/' + urlEncode() + ".status"
```

Comment:
Still not sure what the advantage of this is over, say:

```suggestion
    return directory + '/' + URLEncoder.encode(env.STAGE_NAME, 'UTF-8') + ".status"
```

Does whatever `@NonCPS` does matter for a single operation like `URLEncoder.encode()` or does it really only apply to functions of some non-trivial length?